### PR TITLE
New Blog Post: "Designing a Permanent Personal Identity"

### PR DESCRIPTION
**Pull request by urcades**
Adding this new post, please review at your leisure – I think we want to get it out by tomorrow.

A note:

Seeing the rendering of the footnotes is making me want to tweak how those are styled, but that can be handled in another design pass.

### CODE REVIEW

File: content/blog/pki-maze.md [hunk 1]
```diff
@@ -0,0 +1,72 @@
++++
+title = "Designing a Permanent Personal Identity"
+date = 2019-11-20
```
Comment (matildepark) [inline content/blog/pki-maze.md:3]: Should be 2019-11-27 if going out tomorrow.

File: content/blog/pki-maze.md [hunk 2]
```diff
@@ -0,0 +1,72 @@
...
+
+Thus, we store our PKI data on a blockchain for our second kind of name: planets. A planet is a 32-bit address which has key information stored on the blockchain.[^2]  The owner of a planet may broadcast new PKI data by adding it to the blockchain. Any later messages by the old keys will be rejected, and everyone on the network will listen to the blockchain for key data. Thus, we have global consistency, permanence, and self sovereignty. We know of no other solution that can provide these properties.
+
+However, while small individually, the aggregate PKI data for all nodes on the network may become very large. This is not an issue for comets because nobody needs to store comet keys except for those which they're talking with, and even those can be garbage-collected and re-requested and verified. For planets, there is a canonical set of keys, and somebody must store that. There are about 2^32, or 4 billion planets. If the PKI data is about 100 bytes for each planet, this is about 400 GB of data. This may be more than most users wish to store, but it's small enough that it would be very cheap for someone to host this data for many users[^3].
```
Comment (matildepark) [inline content/blog/pki-maze.md]: 2^32 should use superscript. Which in Markdown is sadly only in HTML. `2<sup>32</sup>`

File: content/blog/pki-maze.md [hunk 3]
```diff
@@ -0,0 +1,72 @@
++++
+title = "Designing a Permanent Personal Identity"
+date = 2019-11-20
+description = "A public key infrastructure (PKI) is a system for binding a set of keys to a name. Sometimes a small amount of metadata is included."
+[extra]
+author = "Philip"
```
Comment (matildepark) [inline content/blog/pki-maze.md]: We tend toward full names on the rest of the blog. Author’s preference of course but don’t we use it elsewhere? In the whitepaper as well?

**Comment by matildepark** [comment]
The footnotes are indeed, “quirky”, but I think a pretty quick fix. I’m curious why footnote links are seemingly being treated as blocks, not inline.

File: content/blog/pki-maze.md [hunk 4]
```diff
@@ -0,0 +1,72 @@
...
+
+In Urbit, a "name" is often called a "ship" or an "address" because we use the metadata in the PKI to make names routable. The total data is two 256-bit asymmetric keys, a cryptographic suite number (to allow changing crypto algorithms), the revision number of the key, and the name of a ship that will route for it. This sums to less than 128 bytes of data.
+
+Each PKI trades off various properties. We chose a tripartite system so that appropriate choices can be made for different use cases. Here, we explore the various properties we chose by following a series of binary choices -- the idea maze.
+
+There are two sorts of PKI: either you can change your keys or not. If you cannot, then your name is impermanent, for noone can keep a set of keys secure forever. Even if your opsec never fails, eventually your crypto algorithms may be compromised.
```
Comment (joemfb) [inline content/blog/pki-maze.md]: "noone" -> "no one"

**Review by joemfb — CHANGES_REQUESTED** [review]
noticed a typo:

File: content/blog/pki-maze.md (hunk 1 above)
Comment (urcades) [reply content/blog/pki-maze.md:3]: good call, will update accordingly when all reviews + edits are addressed

File: content/blog/pki-maze.md (hunk 2 above)
Comment (urcades) [reply content/blog/pki-maze.md]: Fixed!

File: content/blog/pki-maze.md (hunk 3 above)
Comment (urcades) [reply content/blog/pki-maze.md]: Will email him and get his OK

File: content/blog/pki-maze.md (hunk 4 above)
Comment (urcades) [reply content/blog/pki-maze.md]: Great catch, thanks!

File: content/blog/pki-maze.md [hunk 5]
```diff
@@ -0,0 +1,72 @@
++++
+title = "Designing a Permanent Personal Identity"
+date = 2019-11-20
+description = "A public key infrastructure (PKI) is a system for binding a set of keys to a name. Sometimes a small amount of metadata is included."
```
Comment (philipcmonk) [inline content/blog/pki-maze.md:4]: This text should also be in the text of the post, it's important for framing the rest of it.

**Comment by matildepark** [comment]
Oh, that explains a lot. I thought this needed an intro sentence and there it is.

— 
~haddef-sigwen
http://urbit.org

File: content/blog/pki-maze.md (hunk 1 above)
Comment (urcades) [reply content/blog/pki-maze.md:3]: Full name OKed

File: content/blog/pki-maze.md (hunk 5 above)
Comment (urcades) [reply content/blog/pki-maze.md:4]: Ah, just fixed this — thanks for pointing it out.

Comment (philipcmonk) [reply content/blog/pki-maze.md:4]: Come to think of it, for the "description" part, the second sentence is probably noise.

**Review by urcades — COMMENTED** [review]
I'm thinking this is looking great — if we think the image matches the general sentiment of what this essay is conveying, I'm good to merge to master.

File: content/blog/pki-maze.md (hunk 3 above)
Comment (urcades) [reply content/blog/pki-maze.md]: Name OK'd

**Comment by matildepark** [comment]
I'm happy with this (the description could be better but I couldn't find a good pull, myself). I imagine @galenwp wants the last OK?

**Comment by galenwp** [comment]
I trust @philipcmonk. Let's do it.

**Review by galenwp — APPROVED** [review]

**Pull request merged**